src/workerd/io/trace.capnp

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2017-2022 Cloudflare, Inc.
+# Licensed under the Apache 2.0 license found in the LICENSE file or at:
+#     https://opensource.org/licenses/Apache-2.0
+
+@0xc40f73be329a38d9;
+
+using Cxx = import "/capnp/c++.capnp";
+$Cxx.namespace("workerd::rpc");
+
+# This file only contains trace helper structures – due to C++ include size constraints we should
```

Comment:
This was a surprising find – I wanted to have `Trace` itself in here but it turns out that having any kind of capnp interface comes with a significant compile time overhead. This would have increased the compile time of source files using a commonly used downstream capnp file that includes this significantly, undoing some of the progress made through previous include cleanups.

Reply:
Would you mind expanding the comment here to explain what those size constraints are?

Reply:
Absolutely, this is a quite subtle issue so it deserves some more detail.
`.capnp.h` includes are a major driver of slow builds for us – https://github.com/capnproto/capnproto/pull/2188 explores this in-depth and partially mitigates it. This is noticeable especially when compiling code that directly or indirectly depends on many or large capnp structures – `Trace` as defined in worker-interface.capnp is somewhat large and includes capnp interfaces, which cause a lot of code to be generated. This currently causes worker-interface.capnp to have a relatively large include overhead – it's worth paying attention to where Trace is defined to avoid having too many source files needing to transitively include it.
I expanded the comment a bit, hopefully this is more clear now.